### PR TITLE
Polish Our code by YuYang's review

### PR DESCRIPTION
**Pull request by reyoung**

### CODE REVIEW

File: paddle/framework/backward_test.cc [hunk 1]
```diff
@@ -39,9 +39,9 @@ class RowWiseAddOpMaker : public OpProtoAndCheckerMaker {
  public:
   RowWiseAddOpMaker(OpProto *proto, OpAttrChecker *op_checker)
       : OpProtoAndCheckerMaker(proto, op_checker) {
-    AddInput("X", "Input X of Add").IgnoreGradient();
-    AddInput("b", "Bias of Add").IgnoreGradient();
-    AddOutput("Out", "Out of Add").IgnoreGradient();
+    AddInput("X", "Input X of Add").NoGradient();
```
Comment (wangkuiyi) [inline paddle/framework/backward_test.cc]: NoGradient => AsNoGradient
SetIntermediate => AsIntermediate
SetDuplicable => AsDuplicable ?

File: paddle/framework/op_registry_test.cc [hunk 2]
```diff
@@ -51,6 +51,15 @@ class MyTestOpProtoAndCheckerMaker : public OpProtoAndCheckerMaker {
 }  // namespace framework
 }  // namespace paddle
 
+static void ConstructVars(const std::string& param_name,
+                          std::initializer_list<const char*> arguments,
+                          paddle::framework::OpDesc::Var* var) {
+  var->set_parameter(param_name);
+  for (auto& arg_name : arguments) {
+    *var->mutable_arguments()->Add() = arg_name;
```
Comment (wangkuiyi) [inline paddle/framework/op_registry_test.cc]: `var->add_arguments(arg_name)`

File: paddle/framework/op_registry_test.cc [hunk 3]
```diff
@@ -59,13 +68,8 @@ REGISTER_OP(my_test_op, paddle::framework::MyTestOp,
...
   paddle::framework::OpDesc op_desc;
   op_desc.set_type("cos_sim");
-  auto input = op_desc.add_inputs();
-  input->set_parameter("input");
-  *input->mutable_arguments()->Add() = "aa";
-
-  auto output = op_desc.add_outputs();
-  output->set_parameter("output");
-  *output->mutable_arguments()->Add() = "bb";
+  ConstructVars("input", {"aa"}, op_desc.add_inputs());
```
Comment (wangkuiyi) [inline paddle/framework/op_registry_test.cc]: What is the difference between Make, Construct, Create, Build ? Which verb should we use here?

File: paddle/framework/backward_test.cc (hunk 1 above)
Comment (wangkuiyi) [reply paddle/framework/backward_test.cc]: 话说将来弄GAN的时候，如何修改NoGradient这个属性？

Comment (reyoung) [reply paddle/framework/backward_test.cc]: 这个语意，其实不是不需要Gradient的意思。我们开另一个PR来修改这个名字吧。

File: paddle/framework/op_registry_test.cc (hunk 3 above)
Comment (reyoung) [reply paddle/framework/op_registry_test.cc]: By dict [here](http://wikidiff.com/make/build)

"Create" means to bring something into existence.
"Make" means to create. As verbs the difference between make and create is that make is to behave, to act while create is to put into existence. 
"Construct" means to build or form (something) by assembling parts.
"Build" means to form (something) by combining materials or parts.

个人理解:

* "Create" 从无到有
* "Make"  是Create的近义词。
* "Construct" 更倾向于将原有事物组装一些部分。不改变原有事物的本质。
* "Build" 更倾向于将原有事物组合成新的事物，原有事物融合进新的事物中。

这里因为内存不是这个函数申请的，所以不是从无到有的行为。故不使用Create和Make，而这里使用的输入参数会变成新的Var*的一部分。即原有的参数消失，融入进新的Var中。故使用"Build"更合适。

**Comment by reyoung** [comment]
@wangkuiyi Cause this PR is blocking LongFei's work. It just merged now. If `BuildVar` is not a good name, I will give another PR to fix them.

**Pull request merged**